backend-app/app.py

```diff
@@ -0,0 +1,25 @@
+from document_retrieval import DocumentRetrieval
+from answer_generation import PROMPT_TEMPLATE, AnswerGeneration
+import argparse
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Document search and answer generation")
+    parser.add_argument("--query", required=True, help="Enter your query about the ROSA service")
+    args = parser.parse_args()
+    
+    config_path = 'config.json'
+    retrieval = DocumentRetrieval(config_path)
```

Comment:
On initializing this class it looks like we load all of our data into memory, and then either create an in-memory vector database or update an existing one to include the new dataset. Is that right? If so, these seem like expensive operations you'd want to do rarely. Would it be possible to break this up? Or maybe just move some of these more expensive calls out of the `__init__` function?